### PR TITLE
Update .form-check to properly support gradients when enabled

### LINKED ISSUE #29176: Form-Check (Radio, Checkbox & Switch) not displayed correctly, with Form validation & Gradients enabled
Form-Check (Radio, Checkbox & Switch) not displayed correctly, with Form validation & Gradients enabled

grafik
grafik

### PR DESCRIPTION
**Pull request by mdo**
Fixes #29176.

Had to do some updating to a few things to make this work. Not sure how best to clean it up yet, but this at least restores the functionality. Making multiple `background-image`s work without a bunch of duplicate mess is the tricky part here.

- Updates the `gradient-bg()` mixin to include another parameter, a `$foreground` "background image" that is put on top of the gradient.
- Adds a ton of `@if` statements to handle the image swapping.
- Adds a new variable while I'm at it for `$form-switch-bg-size`.

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -565,6 +565,7 @@ $form-switch-width:               2em !default;
 $form-switch-height:              $form-check-input-width !default;
 $form-switch-padding-left:        $form-switch-width + .5em !default;
 $form-switch-bg-image:            url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='-4 -4 8 8'><circle r='3' fill='#{$form-switch-color}'/></svg>") !default;
+$form-switch-bg-size:             calc(#{$form-switch-height} - 2px) !default;
```
Comment (MartijnCuppens) [inline scss/_variables.scss]: Maybe switch to `contain` here?

Comment (mdo) [reply scss/_variables.scss]: `contain`?

Comment (MartijnCuppens) [reply scss/_variables.scss]: `background-size: contain;`, feels a little more flexible.

File: scss/forms/_form-check.scss [hunk 2]
```diff
@@ -105,6 +120,14 @@
     &:checked {
       background-image: escape-svg($form-switch-checked-bg-image);
```
Comment (MartijnCuppens) [inline scss/forms/_form-check.scss]: This can be removed

File: scss/forms/_form-check.scss [hunk 3]
```diff
@@ -103,8 +118,15 @@
...
     &:checked {
-      background-image: escape-svg($form-switch-checked-bg-image);
       background-position: $form-switch-checked-bg-position;
+
+      @if $enable-gradients {
+        background-image: escape-svg($form-switch-checked-bg-image), linear-gradient(180deg, $form-check-input-checked-bg-color, lighten($form-check-input-checked-bg-color, 10%));
+        background-size: $form-switch-bg-size, auto;
+      } @else {
+        background-image: escape-svg($form-switch-checked-bg-image);
+        background-size: $form-switch-bg-size;
```
Comment (MartijnCuppens) [inline scss/forms/_form-check.scss]: This property is also already set (I think this is the last one 😛)

**Review by MartijnCuppens — APPROVED** [review]

**Pull request merged**